pythia/scripts/features/extract_features_vmb.py

```diff
@@ -0,0 +1,179 @@
+# Requires maskrcnn-benchmark to be built and installed
+import argparse
+import glob
+import os
+
+import cv2
+import numpy as np
+import torch
+from PIL import Image
+
+from maskrcnn_benchmark.config import cfg
+from maskrcnn_benchmark.layers import nms
+from maskrcnn_benchmark.modeling.detector import build_detection_model
+from maskrcnn_benchmark.structures.image_list import to_image_list
+from maskrcnn_benchmark.utils.model_serialization import load_state_dict
+from pythia.utils.general import download_file
+
+
+class FeatureExtractor:
+    MODEL_URL = (
+        "https://dl.fbaipublicfiles.com/pythia/detectron_model/detectron_model.pth"
+    )
+    CONFIG_URL = (
+        "https://dl.fbaipublicfiles.com/pythia/detectron_model/detectron_model.yaml"
+    )
+
+    def __init__(self):
+        self.args = self.get_parser().parse_args()
+        self.detection_model = self._build_detection_model()
+
+        if not os.path.exists(self.args.output_folder):
+            os.makedirs(self.args.output_folder, exist_ok=True)
+
+    def _try_downloading_necessities(self):
+        if self.args.model_file is None:
+            print("Downloading model and configuration")
+            self.args.model_file = "./detectron_model.pth"
+            self.args.config_file = "./detectron_model.yaml"
+            download_file(self.MODEL_URL)
+            download_file(self.CONFIG_URL)
+
+    def get_parser(self):
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            "--model_file", default=None, type=str, help="Detectron model file"
```

Comment:
NIT: I have seen the patter `--model-file` to be more commonly used rather than `--model_file`. This is a tiny detail, feel free to ignore it.

Reply:
Would it matter? https://stackoverflow.com/questions/12834785/having-options-in-argparse-with-a-dash They are anyways converted to `_`. Do you know if there is a particular reason people use `-`.